Assets/HoloToolkit/SpatialSound/Scripts/AudioLoFiEffect.cs

```diff
@@ -0,0 +1,222 @@
+
+        // The filters used to simulate the source quality.
+        private AudioLowPassFilter lowPassFilter;
+        private AudioHighPassFilter highPassFilter;
+
+        // Collection used to look up the filter settings that match the selected
+        // source quality.
+        private Dictionary<AudioSourceQuality, FilterSettings> sourceQualityFilterSettings =
+            new Dictionary<AudioSourceQuality, FilterSettings>();
+
+        private void Awake()
+        {
+            LoadQualityFilterSettings();
+
+            filterSettings = sourceQualityFilterSettings[SourceQuality];
+
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+        }
+
+        private void Update()
+        {
+            FilterSettings fs = sourceQualityFilterSettings[SourceQuality];
+            if (fs != filterSettings)
+            {
+                // If we have an attached AudioInfluencerManager, we need to let it know
+                // about our filter settings change.
+                AudioInfluencerManager influencerManager = gameObject.GetComponent<AudioInfluencerManager>();
+                if (influencerManager != null)
+                {
+                    influencerManager.SetNativeLowPassCutoffFrequency(fs.LowPassCutoff);
+                    influencerManager.SetNativeHighPassCutoffFrequency(fs.HighPassCutoff);
+                }
+
+                filterSettings = fs;
+                lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+                highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+            }
+        }
+
+        // Populates the source quality filter settings collection.
+        private void LoadQualityFilterSettings()
+        {
+            if (sourceQualityFilterSettings.Keys.Count > 0) { return; }
+
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.FullRange,
+                new FilterSettings(10, 22000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.NarrowBandTelephone,
+                new FilterSettings(300, 3400));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.WideBandTelephone,
+                new FilterSettings(50, 7000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.AmRadio,
+                new FilterSettings(40, 5000));
+            sourceQualityFilterSettings.Add(
+                AudioSourceQuality.FmRadio,
+                new FilterSettings(30, 15000));
+        }
+
+        /// <summary>
+        /// Source quality options that match common telephony and
+        /// radio broadcast options.
+        /// </summary>
+        public enum AudioSourceQuality
```

Comment:
Please extract enum in to it's own file.  We should never embed enums unless they are private

Reply:
I'll extract the enum and the struct from the class. They are specific to working with this class, so I'll leave them in the source file, but make their names clearly imply that they are for use with the AudioLoFiEffect.